src/celeritas/io/ImportOpticalModel.hh

```diff
@@ -0,0 +1,56 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2024 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file celeritas/io/ImportOpticalModel.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include <vector>
+
+#include "ImportPhysicsVector.hh"
+
+namespace celeritas
+{
+namespace optical
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Enumerator for available optical physics models.
+ *
+ * This enum is used to identify the optical model that imported model MFP
+ * tables correspond to.
+ */
+enum class ImportModelClass
+{
+    other,
+    absorption,
+    rayleigh,
+    wls,
+    //
+    size_
+};
+
+//---------------------------------------------------------------------------//
+// FREE FUNCTIONS
+//---------------------------------------------------------------------------//
+
+// Get the string form of one of the enumerations
+char const* to_cstring(ImportModelClass imc);
+
+//---------------------------------------------------------------------------//
+}  // namespace optical
+
+//---------------------------------------------------------------------------//
+/*!
+ * Imported data for an optical physics model.
+ */
+struct ImportOpticalModel
+{
+    optical::ImportModelClass model_class;
+    std::vector<ImportPhysicsVector> mfps;  //!< per optical material MFPs
```

Comment:
This is redundant with the `ImportOpticalMaterial`.{rayleigh,absorption,wls} data which already tells us the available models and stores `absorption_length` (mfp): should we embed model information into the material data, or have separate model classes?
```c++
/*!
 * Store optical material properties.
 */
struct ImportOpticalMaterial
{
    ImportPhysicsVector refractive_index;
    ImportScintData scintillation;
    bool cerenkov{false};

    //! Whether minimal useful data is stored
    explicit operator bool() const
    {
        return static_cast<bool>(refractive_index);
    }
};

struct ImportOpticalWLSModel
{
    ImportPhysicsVector absorption_length;  //!< Absorption length [MeV, len]

    double mean_num_photons{};  //!< Mean number of re-emitted photons
    double time_constant{};  //!< Time delay between absorption and re-emission
    ImportPhysicsVector component;  //!< Re-emission population [MeV, unitless]

    explicit operator bool() const
    {
        return mfp && mean_num_photons > 0 && time_constant > 0
               && static_cast<bool>(absorption_length)
               && static_cast<bool>(component)
               && absorption_length.vector_type == ImportPhysicsVectorType::free
               && component.vector_type == absorption_length.vector_type;
    }
};

struct ImportOpticalAbsorptionModel
{
    ImportPhysicsVector absorption_length;  //!< Absorption length [MeV, len]

    explicit operator bool() const
    {
        return static_cast<bool>(absorption_length);
    }
};
```